app/assets/javascripts/Components/Result/result.jsx

```diff
@@ -27,6 +27,13 @@ class Result extends React.Component {
       submission_files: {files: [], directories: {}, name: '', path: []},
       fullscreen: false,
       annotationModal: INITIAL_ANNOTATION_MODAL_STATE,
+      props: {
```

Comment:
It actually is a bit confusing to use "props" as a state key. Better to flatten: keep the "_id" parts as just top-level keys in the state, and actually `role` can remain a prop, it doesn't need to be a state since it'll never change.